packages/components/src/styles/components/modal.scss

```diff
@@ -101,26 +58,19 @@
   width: min(800px, 95vw);
 }
 
-.hds-modal--color-neutral {
-  .hds-modal__header {
-    color: var(--token-color-foreground-strong);
-    background: var(--token-color-surface-faint);
-    border-bottom: 1px solid var(--token-color-border-primary);
-  }
-
-  .hds-modal__tagline {
-    color: var(--token-color-foreground-faint);
-  }
-}
+// Note ".hds-modal--color-neutral" uses default color and background-color styles from the dialog-primitive
 
 .hds-modal--color-warning {
```

Comment:
General idea: what if we use dedicated (local) CSS variables, instead of overriding colors using CSS selectors? In this way we don't enter the weeds of using `dialog-primitive` selectors inside a `modal` CSS file. See for example how we're doing for the Dropdown component https://github.com/hashicorp/design-system/blob/main/packages/components/src/styles/components/dropdown.scss#L423-L454 (and we should do also in other parts too, eg. the Alert component; but that's another story for the future :) )

Reply:
I will look at this more closely, thanks